merge.py

```diff
@@ -1,98 +1,320 @@
+                self.work_queue.put(pr)
+
+    def merge_git_pr(self, pr):
+        """merge_git_pr merges git pull requests.
+
+        Args:
+            pr: The GithubPR to merge.
+        Returns:
+            True if merge was concluded satisfactorily (merged successfully,
+            or failed due to supposed fault of the PR itself).
+            False if failure was due to an environmental issue and should be
+            retried.
+        """
+        repo = self.config['repository']
+        tmp_dir = TMP_DIR_FMT.format(dir='{}-{}'.format(repo, pr.get_num()))
+        try:
+            _set_up(tmp_dir)
+        except AssertionError:
+            pr.post_error('Error setting up - trying again.')
+            return False
+        try:
+            self.clone_configure(tmp_dir, pr)
+            pr_name = 'finish-pr-{}'.format(pr.get_num())
+            self.checkout(pr_name, tmp_dir, pr)
+            branch = '{}/{}'.format('apache', self.config['merge_branch'])
+            self.rebase(branch, tmp_dir, pr)
+            self.target_checkout(branch, tmp_dir, pr)
+            self.do_local_merge(pr_name, tmp_dir, pr)
+            self.verify(tmp_dir, pr)
+            self.push(tmp_dir, pr)
+            _clean_up(tmp_dir)
+            _print_flush('Merged successfully!')
+        except AssertionError as err:
+            _print_flush(err)
+        except EnvironmentError as err:
+            _print_flush(err)
+            return False
+        return True
+
+    def clone_configure(self, tmp_dir, pr):
+        """Clone repository and configure.
+
+        Args:
+            tmp_dir: Directory in which commands should be run.
+            pr: Pull request to merge.
+        Raises:
+            AssertionError: If command was not successful.
+            EnvironmentError: If PR comment couldn't be posted to Github.
+        """
+        _print_flush('Cloning repository and configuring.')
+        repo = self.config['repository']
+        repo_url = GITHUB_REPO_URL.format(org='apache', repo=repo)
+        clone_args = ['git', 'clone', '-b', self.config['merge_branch'],
+                      repo_url, tmp_dir]
+        try:
+            check_call(clone_args, cwd=tmp_dir)
+        except:
+            pr.post_error(
+                'Couldn\'t clone from github/{}/{}. Please try again.'.format(
+                    'apache', repo))
+            raise AssertionError('Clone Failed.')
+
+        try:
+            check_call(['git', 'remote', 'add', 'apache',
+                        APACHE_GIT.format(repo=repo)], cwd=tmp_dir)
+            check_call(['git', 'remote', 'rename', 'origin', 'github'],
+                       cwd=tmp_dir)
+            check_call('git config --local --add remote.github.fetch'
+                       ' "+refs/pull/*/head:refs/remotes/github/pr/*"',
+                       shell=True, cwd=tmp_dir)
+            check_call(['git', 'fetch', '--all'], cwd=tmp_dir)
```

Comment:
The commands here are taken from Davor's merge script. I don't know if he had a reason for fetching all; I'll ask him.